### PR TITLE
Add polygon mooar nft models

### PR DESCRIPTION
**Pull request by BennyFeng**
# Thank you for contributing to Spellbook!
Please refer to the top of the `readme` in the root of Spellbook to learn how to contribute to Spellbook on DuneSQL.

### CODE REVIEW

**Comment by springzh** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by springzh** [comment]
@Hosuke @jeff-dude We're working on some NFT marketplaces for Polygon. The Mooar marketplace is a fork of Seaport v1.5. To reduce duplicated code, we are trying to reuse the `seaport_v4_trades` macro. Since we need different contract address, so there is some modification to the macro. Please review and advise. Thank you.

**Comment by jeff-dude** [comment]
> @Hosuke @jeff-dude We're working on some NFT marketplaces for Polygon. The Mooar marketplace is a fork of Seaport v1.5. To reduce duplicated code, we are trying to reuse the `seaport_v4_trades` macro. Since we need different contract address, so there is some modification to the macro. Please review and advise. Thank you.

love the usage of the macro(s)!

i've tagged @0xRobin for a quick review, but i think leveraging those is the correct approach.

File: macros/models/_sector/nft/platforms/seaport_v4_trades.sql
Comment (0xRobin) [inline macros/models/_sector/nft/platforms/seaport_v4_trades.sql:1]: There is a `seaport_fork_trades()` macro that does exactly what your trying to do here, but for seaport v3.
(adapt the seaport macro to also support deployed forks)

Can you remove the changes from this macro and move this to a new (`seaport_v4_fork_trades`) macro, might be best to also do a name change for `seaport_fork_trades` -> `seaport_v3_fork_trades` to make the version distinction clear?  
You will need to add parameters to pass in the project and project version.

**Review by 0xRobin — CHANGES_REQUESTED** [review]

File: macros/models/_sector/nft/platforms/seaport_v4_trades.sql
Comment (0xRobin) [reply macros/models/_sector/nft/platforms/seaport_v4_trades.sql:1]: The main reason I went with this setup is to keep the seaport macro's clear of any fork logic
(they are already complex enough on their own)

Comment (springzh) [reply macros/models/_sector/nft/platforms/seaport_v4_trades.sql:1]: Sounds good. We'll create a seaport_v4_fork_trades and handle rename for seaport_v3_fork_trades. Thanks.

Comment (BennyFeng) [reply macros/models/_sector/nft/platforms/seaport_v4_trades.sql:1]: Updated.

**Review by 0xRobin — APPROVED** [review]
thanks for incorporating the changes, this looks good now! :ok_hand:

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**